src/read-only-handler.ts

```diff
@@ -42,7 +42,11 @@ export class ReadOnlyHandler {
             const { originalTarget } = this;
             throw new Error(`Invalid mutation: Cannot set "${key.toString()}" on "${originalTarget}". "${originalTarget}" is read-only.`);
```

Comment:
@caridy What do you think about removing all the if `process.env.NODE_ENV !== 'production'` checks to provide a consistent error message on all the engines? 
We can make share the same error message for all the handlers to reduce code size even in production mode.

Reply:
this is a bra**king change in LWC since we don't throw in prod today due to the build step to eliminate this code-branch.

Reply:
Regardless of the mode, the read-only membrane throws because the Proxy trap returns `false`. When running in production mode, the error is just a little more cryptic than the one in dev mode.